integration-tests/scripts/copy-branch-to-repo-git.sh

```diff
@@ -0,0 +1,207 @@
+#   GITHUB_TOKEN        - A GitHub personal access token with repo creation and push permissions. Required.
+#
+# Dependencies:
+#   git, curl, jq, create-github-repo.sh (must be in same directory)
+#
+# Note:
+#   This script clones the source repo, adds the destination as a remote, and pushes
+#   the branch directly. This works regardless of whether repos share history.
+#   If the destination repository doesn't exist, it will be created automatically as a public repository.
+
+set -eo pipefail
+
+# Enable debug mode if DEBUG environment variable is set
+if [ -n "${DEBUG}" ]; then
+  set -x
+  echo "🐛 Debug mode enabled"
+fi
+
+if [ -z $GITHUB_TOKEN ] ; then
+  echo "🔴 error: missing env var GITHUB_TOKEN"
+  exit 1
+fi
+
+source_repo=$1
+if [ -z "$source_repo" ] ; then
+  echo "🔴 error: missing parameter source_repo"
+  exit 1
+fi
+
+source_branch=$2
+if [ -z "$source_branch" ] ; then
+  echo "🔴 error: missing parameter source_branch"
+  exit 1
+fi
+
+dest_repo=$3
+if [ -z "$dest_repo" ] ; then
+  echo "🔴 error: missing parameter dest_repo"
+  exit 1
+fi
+
+dest_branch=$4
+if [ -z "$dest_branch" ] ; then
+  echo "🔴 error: missing parameter dest_branch"
+  exit 1
+fi
+
+# Verify source repository exists and is accessible
+echo "Verifying source repository ${source_repo} exists and is accessible..."
+SOURCE_REPO_RESPONSE=$(curl -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/repos/${source_repo} 2> /dev/null)
+if [ -n "${DEBUG}" ]; then
+  echo "🐛 Source repo API response: ${SOURCE_REPO_RESPONSE}"
+fi
+SOURCE_REPO_CHECK=$(echo "${SOURCE_REPO_RESPONSE}" | jq -r '.full_name // ""')
+if [ -z "${SOURCE_REPO_CHECK}" ]; then
+  echo "🔴 error: source repository ${source_repo} not found or not accessible"
+  if [ -n "${DEBUG}" ]; then
+    echo "🐛 Full API response: ${SOURCE_REPO_RESPONSE}"
+  fi
+  echo "   Check that:"
+  echo "   1. Repository name is correct"
+  echo "   2. Repository exists"
+  echo "   3. GITHUB_TOKEN has access to read the repository"
+  exit 1
+fi
+
+# Verify destination repository exists and is accessible
+echo "Verifying destination repository ${dest_repo} exists and is accessible..."
+DEST_REPO_RESPONSE=$(curl -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/repos/${dest_repo} 2> /dev/null)
+if [ -n "${DEBUG}" ]; then
+  echo "🐛 Destination repo API response: ${DEST_REPO_RESPONSE}"
+fi
+DEST_REPO_CHECK=$(echo "${DEST_REPO_RESPONSE}" | jq -r '.full_name // ""')
+if [ -z "${DEST_REPO_CHECK}" ]; then
+  echo "⚠️  Destination repository ${dest_repo} not found"
+  echo "   Attempting to create it automatically..."
+  
+  # Pass the full repository path to the create script (it now handles org/repo format)
+  if [ -n "${DEBUG}" ]; then
+    echo "🐛 Full dest_repo: ${dest_repo}"
+  fi
+  
+  # Get the directory where this script is located to find the helper script
+  SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+  CREATE_REPO_SCRIPT="${SCRIPT_DIR}/create-github-repo.sh"
```

Comment:
This seems like overkill. We have 15 lines to call this script. I don't think we have to check if the script exists. Call it, if $? -ne 0, say the error. We don't need most of lines 100-113